core/src/test/java/org/infinispan/test/AbstractInfinispanTest.java

```diff
@@ -72,6 +113,191 @@
+
+         // This shouldn't block long when held - but it is a write lock which can be delayed
+         builder.allowBlockingCallsInside(KeyAffinityServiceImpl.class.getName(), "handleViewChange");
+
+         // The cache has to always be LOCAL and cannot be used with persistence
+         builder.allowBlockingCallsInside(RecoveryManagerImpl.class.getName(), "registerInDoubtTransaction");
+
+         // Allow logging to block
+         builder.allowBlockingCallsInside(Logger.class.getName(), "logMessage");
+         builder.allowBlockingCallsInside(java.util.logging.Logger.class.getName(), "log");
+
+         // Make sure our rejection handlers never block in a non blocking thread
+         builder.disallowBlockingCallsInside(BlockingRejectedExecutionHandler.class.getName(), "rejectedExecution");
+
+         builder.allowBlockingCallsInside(AbstractInfinispanTest.class.getName(), "writeJUnitReport");
+
+         methodsToBeRemoved(builder);
+         jgroupsMethodsAllowedToBlock(builder);
+
+         questionableMethodsAllowedToBlock(builder);
+
+         allowTestsToBlock(builder);
+
+         builder.blockingMethodCallback(bm ->
+            writeJUnitReport(new AssertionError(String.format("Blocking call! %s on thread %s", bm, Thread.currentThread())), "Blocking"));
+      }).install();
+
+      Thread.setDefaultUncaughtExceptionHandler((thread, t) -> {
+         LogFactory.getLogger("Infinispan-TEST").fatal("Throwable was not caught in thread " + thread +
+               " - exception is: " + t);
+         // RxJava propagates via this and we don't want to worry about it
+         if (!(t instanceof UndeliverableException)) {
+            writeJUnitReport(t, "Uncaught");
+         }
+      });
+   }
+
+   private static void methodsToBeRemoved(BlockHound.Builder builder) {
+      // Total Order is to be removed!
+      builder.allowBlockingCallsInside(StateConsumerImpl.class.getName(), "awaitTotalOrderTransactions");
+
+      // The internal map only supports local mode - we need to replace with Caffeine
+      builder.allowBlockingCallsInside(RecoveryManagerImpl.class.getName(), "registerInDoubtTransaction");
+   }
+
+   private static void jgroupsMethodsAllowedToBlock(BlockHound.Builder builder) {
+      // UDP calls PlainDatagramSocketImpl#send which can block if OS buffer is full - but we just ignore
+      builder.allowBlockingCallsInside(UDP.class.getName(), "_send");
+      builder.allowBlockingCallsInside(FlowControl.class.getName(), "adjustCredit");
+      builder.allowBlockingCallsInside(Table.class.getName(), "add");
+      builder.allowBlockingCallsInside(Table.class.getName(), "get");
+
+      // THIS IS A VERY BAD HACK - because you can't instrument a static block of the class in jgroups Version class
+      // which loads a properties file
+      builder.allowBlockingCallsInside(Properties.class.getName(), "load");
```

Comment:
Couldn't you reference `org.jgroups.Version` here instead, so it's loaded before Blockhound is installed?

Reply:
I hadn't thought about this, but I like it. I can do that instead :)